packages/desktop/components/popups/WithdrawFromL2Popup.svelte

```diff
@@ -0,0 +1,170 @@
+<script lang="ts">
+    import { PopupId, closePopup, openPopup } from '@auxiliary/popup'
+    import { getSelectedAccount } from '@core/account'
+    import { localize } from '@core/i18n'
+    import { getArchivedBaseTokens } from '@core/layer-2/helpers/getArchivedBaseTokens'
+    import { getBaseToken, getCoinType, activeProfile, isActiveLedgerProfile } from '@core/profile'
+    import { truncateString } from '@core/utils'
+    import { formatTokenAmountPrecise, getRequiredStorageDepositForMinimalBasicOutput } from '@core/wallet'
+    import { Button, FontWeight, KeyValueBox, Spinner, Text, TextType } from 'shared/components'
+    import { onMount } from 'svelte'
+    import { getLayer2WithdrawRequest } from '@core/layer-2/utils'
+    import { withdrawL2Funds } from '@core/layer-2/helpers/widthdrawL2Funds'
+    import { getL2ReceiptByRequestId } from '@core/layer-2/helpers/getL2ReceiptByRequestId'
+    import { showAppNotification } from '@auxiliary/notification'
+    import { Bip44 } from '@iota/sdk/out/types'
+    import { displayNotificationForLedgerProfile, ledgerNanoStatus } from '@core/ledger'
+    import { getEstimatedGasForOffLedgerRequest, getNonceForWithdrawRequest } from '@core/layer-2/helpers'
+
+    export let password: string
+    export let withdrawableAmount: number
+
+    const bip44Chain: Bip44 = {
+        coinType: Number(getCoinType()),
+        account: getSelectedAccount().index,
+        change: 0,
+        addressIndex: 0,
+    }
+
+    let address: string | undefined = undefined
+    let isWithdrawing = false
+    const { isStrongholdLocked } = $activeProfile
+
+    $: password && address && !$isStrongholdLocked && withdrawFromL2()
+
+    function onCancelClick(): void {
+        closePopup()
+    }
+
+    function onWithdrawFromL2Click(): void {
+        openUnlockStrongholdPopup()
```

Comment:
```suggestion
        if ($isSoftwareProfile && $isStrongholdLocked) {
        openUnlockStrongholdPopup()
        } 
```
We would only have to unlock the stronghold, if it is previously locked